src/Orleans.Runtime/Utilities/ImmutableCollectionExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Immutable;
+
+namespace Orleans.Runtime.Utilities
+{
+    internal static class ImmutableCollectionExtensions
+    {
+        public static int FindIndex<T>(this ImmutableArray<T> collection, Func<T, bool> predicate)
+        {
+            var index = 0;
+            foreach (var item in collection)
+            {
+                if (predicate(item)) return index;
```

Comment:
This will always return 0. Was the intent here to return index of the item that satisfies the predicate?

Reply:
Oh that's terrible, can you submit a fix?

Reply:
Sure

Reply:
How embarassing

Reply:
Need more sleep? 😉

Reply:
Let's chalk it up to that 😅

Reply:
I wonder how many issues this caused in previous tests (this code is all based on the previous bits which were reverted) 🤔🤔🤔